### PR TITLE
Support for setting (and updating) router debug mode

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Review by fgiorgetti — COMMENTED** [review]
I have created a site controlled Skupper instance and I was able to set, modify and remove
both router debug mode and router logging level.

Using skupper CLI I was able to set the router debug mode properly (as well as logging).
I was also able to remove the debug mode by removing the QDROUTERD_DEBUG env var from the skupper-router deployment.

The only thing that I am missing is the ability to modify it through CLI, when not using site-controller. Should I create an issue to propose the ability to control these two parameters
using CLI, or is it out of scope?

**Comment by grs** [comment]
> The only thing that I am missing is the ability to modify it through CLI,

There should be a new option --router-debug-mode on skupper init. https://github.com/skupperproject/skupper/pull/399/files#diff-117f3c06eb49c3bea90449383ac293b1ace71878fdf08287b9c6ddf83d5d04b2R266

File: client/router_create.go [hunk 1]
```diff
@@ -368,6 +368,12 @@ func (cli *VanClient) GetRouterSpecFromOpts(options types.SiteConfigSpec, siteId
 		Name:  "SKUPPER_SITE_ID",
 		Value: siteId,
 	})
+	if options.RouterDebugMode != "" {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  "QDROUTERD_DEBUG",
```
[the remark this answers is not in the document]
Comment (grs) [reply client/router_create.go:373]: Sorry! I have not checked that in yet.

**Comment by fgiorgetti** [comment]
> > The only thing that I am missing is the ability to modify it through CLI,
> 
> There should be a new option --router-debug-mode on skupper init. https://github.com/skupperproject/skupper/pull/399/files#diff-117f3c06eb49c3bea90449383ac293b1ace71878fdf08287b9c6ddf83d5d04b2R266

Yes and it works just fine. The issue is once you set it to use gdb (in example), and then you want
to stop using the debug mode. In this case CLI won't help and you will need to manually change
the skupper-router deployment and the skupper-site cm.

**Comment by grs** [comment]
>The issue is once you set it to use gdb (in example), and then you want
> to stop using the debug mode.

You can just re-run skupper init with the empty string as the debug mode (which is actually the default)

**Comment by fgiorgetti** [comment]
> > The issue is once you set it to use gdb (in example), and then you want
> > to stop using the debug mode.
> 
> You can just re-run skupper init with the empty string as the debug mode (which is actually the default)

Nice! Good to know. And by the way, it worked just fine as well.

**Pull request merged**